### PR TITLE
Cleanup adjust_float_expressions.cpp

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 88e527b435f03fc816bf899c93e389e83e3fb24b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84806585

File: src/goto-programs/adjust_float_expressions.cpp [hunk 1]
```diff
@@ -98,7 +98,9 @@ void adjust_float_expressions(exprt &expr, const exprt &rounding_mode)
       if(expr.operands().size()>2)
         expr=make_binary(expr);
 
-      assert(expr.operands().size()==2);
+      INVARIANT(
+        expr.operands().size()==2,
```
Comment (peterschrammel) [inline src/goto-programs/adjust_float_expressions.cpp]: clang-format

File: src/goto-programs/adjust_float_expressions.cpp [hunk 2]
```diff
@@ -98,7 +98,9 @@ void adjust_float_expressions(exprt &expr, const exprt &rounding_mode)
...
+        "arithmetic operations must be representable as a binary operation");
```
Comment (peterschrammel) [inline src/goto-programs/adjust_float_expressions.cpp]: I think this is float-specific?

File: src/goto-programs/adjust_float_expressions.cpp [hunk 3]
```diff
@@ -81,8 +81,8 @@ void adjust_float_expressions(exprt &expr, const exprt &rounding_mode)
     return;
 
   // recursive call
-  for(auto &op : expr.operands())
-    adjust_float_expressions(op, rounding_mode);
+  Forall_operands(op, expr)
```
Comment (peterschrammel) [inline src/goto-programs/adjust_float_expressions.cpp]: What's the reason for changing back to the old macros?

File: src/goto-programs/adjust_float_expressions.cpp (hunk 2 above)
Comment (kroening) [reply src/goto-programs/adjust_float_expressions.cpp]: No; this has two parts: a) there is an invariant that +, * etc have two operands or more; b) it is a post-condition of make_binary that you get no more than two operands. I'd add a DATA_INVARIANT for a) and a CHECK_RETURN for b).

File: src/goto-programs/adjust_float_expressions.cpp (hunk 3 above)
Comment (chrisr-diffblue) [reply src/goto-programs/adjust_float_expressions.cpp]: Just consistency with the rest of the file - are we incrementally moving to range-based for loops now then? If so, I'll drop this hunk.

File: src/goto-programs/adjust_float_expressions.cpp (hunk 2 above)
Comment (chrisr-diffblue) [reply src/goto-programs/adjust_float_expressions.cpp]: Thanks for that explanation @kroening - that's much clearer, and wasn't obvious to me from the code as originally written. I'll updated accordingly, and agree that DATA_INVARIANT and CHECK_RETURN are the way to go.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 342784885b8a8bd09175d96d754bd14959331de3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85271341

**Review by tautschnig — APPROVED** [review]

**Pull request merged**